AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -119,4 +127,12 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
     private func createMeasurementStream(for sensorName: MeasurementStreamSensorName, sensorPackageName: String) -> MeasurementStream {
         MeasurementStream(sensorName: sensorName, sensorPackageName: sensorPackageName)
     }
+    
+    private func getExistingSession(sessionUUID: SessionUUID) -> SessionEntity? {
+        var session: SessionEntity?
+        measurementStreamStorage.accessStorage { storage in
+            session = try? storage.getExistingSession(with: sessionUUID)
+        }
+        return session
```

Comment:
It's not safe to return CoreData entities outside of it's NSManagedObjectContext queue

Reply:
Got it, changes everything to `completion`. Not sure if it can be than easier and I understand it is not the prettiest solution :.